contracts/staking/test/unit_tests/mixin_stake_storage_test.ts

```diff
@@ -0,0 +1,154 @@
+            expect(actualBalances[0]).to.deep.equal({
+                ...defaultSyncedBalance,
+                nextEpochBalance: defaultSyncedBalance.nextEpochBalance.minus(amount),
+            });
+            expect(actualBalances[1]).to.deep.equal({
+                ...defaultSyncedBalance,
+                nextEpochBalance: defaultSyncedBalance.nextEpochBalance.plus(amount),
+            });
+        });
+        it('Noop if pointers are equal', async () => {
+            await testContract.setStoredBalance.awaitTransactionSuccessAsync(defaultSyncedBalance, INDEX_ZERO);
+            // If the pointers weren't equal, this would revert with InsufficientBalanceError
+            await testContract.moveStake.awaitTransactionSuccessAsync(
+                INDEX_ZERO,
+                INDEX_ZERO,
+                defaultSyncedBalance.nextEpochBalance.plus(1),
+            );
+            const actualBalance = await getTestBalancesAsync(INDEX_ZERO);
+            expect(actualBalance).to.deep.equal(defaultSyncedBalance);
+        });
+        it("Reverts if attempting to move more than next epoch's balance", async () => {
+            await testContract.setStoredBalance.awaitTransactionSuccessAsync(defaultSyncedBalance, INDEX_ZERO);
+            const amount = defaultSyncedBalance.nextEpochBalance.plus(1);
+            const tx = testContract.moveStake.awaitTransactionSuccessAsync(INDEX_ZERO, INDEX_ONE, amount);
+            await expect(tx).to.revertWith(
+                new StakingRevertErrors.InsufficientBalanceError(amount, defaultSyncedBalance.nextEpochBalance),
+            );
+        });
+    });
+
+    describe('Load balance', () => {
+        it('_loadSyncedBalance does not change state if balance was previously synced in the current epoch', async () => {
+            await testContract.setStoredBalance.awaitTransactionSuccessAsync(defaultSyncedBalance, INDEX_ZERO);
+            const actualBalance = await testContract.loadSyncedBalance.callAsync(INDEX_ZERO);
+            expect(actualBalance).to.deep.equal(defaultSyncedBalance);
+        });
+        it('_loadSyncedBalance updates current epoch fields if the balance has not yet been synced in the current epoch', async () => {
+            await testContract.setStoredBalance.awaitTransactionSuccessAsync(defaultUnsyncedBalance, INDEX_ZERO);
+            const actualBalance = await testContract.loadSyncedBalance.callAsync(INDEX_ZERO);
+            expect(actualBalance).to.deep.equal(defaultSyncedBalance);
+        });
+        it('_loadUnsyncedBalance loads balance from storage without synchronizing fields', async () => {
+            await testContract.setStoredBalance.awaitTransactionSuccessAsync(defaultUnsyncedBalance, INDEX_ZERO);
+            const actualBalance = await testContract.loadUnsyncedBalance.callAsync(INDEX_ZERO);
+            expect(actualBalance).to.deep.equal(defaultUnsyncedBalance);
+        });
+    });
+
+    describe('Increment/decrement balance', () => {
```

Comment:
Optional: It would be nice to add tests for cases when the `SafeMath` should fail. This is good prevention against regressions.